dali/kernels/tensor_shape.h

```diff
@@ -510,6 +515,7 @@ struct TensorListShapeBase {
 
   decltype(shapes.data()) data() { return shapes.data(); }
   constexpr bool empty() const { return size() == 0; }
+  int size() const { return num_samples_; }
```

Comment:
Do we need `size` if `num_samples` is already here?

Reply:
Which one should we use?
TensorShape and TensorListShape have size() for number of elements and number of samples correspondingly.
Tensor and TensorList use size() from dali::Buffer to return the number of elements (allocation).
And TensorList has ntensor() for obtaining the number of samples. Should we unify that somehow?

Reply:
If we decide on something, I will do a follow-up.